stdlib/list.ml

```diff
@@ -56,6 +56,14 @@ let rec rev_append l1 l2 =
 
 let rev l = rev_append l []
 
+let init n f =
+  if n = 0 then [] else
+  if n < 0 then invalid_arg "List.init" else
+  let rec init_aux acc i =
```

Comment:
Isn't this going to allocate the closure for `init_aux` for each call to init (without flamba)? What about:

```
let init_aux acc i f = 
....
;;

let init n f =
   ... (* checks *);
   rev (init_aux [] 0 f)
;;
```